C#/Botkit/botbuilder-slack-adapter/SlackDialog.cs

```diff
@@ -1,267 +1,267 @@
-        {
-            DialogElement element;
-
-            element = new DialogElement
-            {
-                Label = label,
-                Name = name,
-                Value = value,
-                Type = "select",
-                OptionList = optionList,
-            };
-
-            if (options.GetType().ToString() == "DialogElement")
-            {
-                element = (DialogElement)options;
-            }
-
-            data.Elements.Add(element);
-        }
-
-        /// <summary>
-        /// Get the dialog object as a JSON encoded string. 
-        /// </summary>
-        public string AsString()
-        {
-            return JsonConvert.ToString(this.data.ToString());
-        }
-
-        /// <summary>
-        /// Get the dialog object for use with bot.replyWithDialog() 
-        /// </summary>
-        public object AsObject()
-        {
-            return data;
-        }
-    }
-}
+using System.Collections;
+using System.Collections.Generic;
+
+namespace botbuilder_slack_adapter
+{
+    public class SlackDialog
+    {
+        private DialogData data;
+        private readonly List<DialogElement> elements;
+
+        /// <summary>
+        /// Create a Slack Dialog object for use with [replyWithDialog()](#replyWithDialog)
+        /// </summary>
+        /// <param name="title">The title of the dialog.</param>
+        /// <param name="callbackId">Callback id of dialog.</param>
+        /// <param name="submitLabel">Label for the submit button.</param>
+        /// <param name="element">An array of dialog elements.</param>
+        public SlackDialog(string title, string callbackId, string submitLabel, List<DialogElement> elements)
+        {
+            data = new DialogData(title, callbackId, submitLabel, elements);
+        }
+
+        /// <summary>
+        /// Set the State property of the dialog
+        /// </summary>
+        /// <param name="value">Value for state.</param>
+        public void SetState(string value)
+        {
+            data.State = value;
+        }
+        
+        /// <summary>
+        /// Set the NotifyOnCancel property of the dialog
+        /// </summary>
+        /// <param name="set">Set true to have Slack notify you with a `dialog_cancellation` event if a user cancels the dialog without submitting.</param>
+        public void SetNotifyOnCancel(bool set)
+        {
+            data.NotifyOnCancel = set;
+        }
+
+        /// <summary>
+        /// Set the title of the dialog
+        /// </summary>
+        /// <param name="value">Value for title.</param>
+        public void SetTitle(string value)
+        {
+            data.Title = value;
+        }
+
+        /// <summary>
+        /// Set the dialog's callback_id
+        /// </summary>
+        /// <param name="value">Value for the callback_id.</param>
+        public void SetCallbackId(string value)
+        {
+            data.Title = value;

```

Comment:
Here should we set the CallBackId property instead of the Title again?

Reply:
@ceciliaavila Fixed!